resources/dynamodb_tables.go

```diff
@@ -0,0 +1,714 @@
+	})
+}
+func resolveDynamodbTableKeySchema(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(*types.TableDescription)
+	return resource.Set(c.Name, marshalKeySchema(r.KeySchema))
+}
+func resolveDynamodbTableRestoreSummary(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(*types.TableDescription)
+	if r.RestoreSummary == nil {
+		return nil
+	}
+	return resource.Set(c.Name, map[string]interface{}{
+		"date_time":         r.RestoreSummary.RestoreDateTime,
+		"in_progress":       r.RestoreSummary.RestoreInProgress,
+		"source_table_arn":  r.RestoreSummary.SourceTableArn,
+		"source_backup_arn": r.RestoreSummary.SourceBackupArn,
+	})
+}
+func resolveDynamodbTableStreamSpecification(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(*types.TableDescription)
+	if r.StreamSpecification == nil {
+		return nil
+	}
+	return resource.Set(c.Name, map[string]interface{}{
+		"enabled":   r.StreamSpecification.StreamEnabled,
+		"view_type": r.StreamSpecification.StreamViewType,
+	})
+}
+func fetchDynamodbTableGlobalSecondaryIndexes(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	p := parent.Item.(*types.TableDescription)
+	for i := range p.GlobalSecondaryIndexes {
+		res <- p.GlobalSecondaryIndexes[i]
+	}
+	return nil
+}
+func resolveDynamodbTableGlobalSecondaryIndexKeySchema(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.GlobalSecondaryIndexDescription)
+	return resource.Set(c.Name, marshalKeySchema(r.KeySchema))
+}
+func fetchDynamodbTableLocalSecondaryIndexes(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	p := parent.Item.(*types.TableDescription)
+	for i := range p.LocalSecondaryIndexes {
+		res <- p.LocalSecondaryIndexes[i]
+	}
+	return nil
+}
+func resolveDynamodbTableLocalSecondaryIndexKeySchema(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.LocalSecondaryIndexDescription)
+	return resource.Set(c.Name, marshalKeySchema(r.KeySchema))
+}
+func fetchDynamodbTableReplicas(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	p := parent.Item.(*types.TableDescription)
+	for i := range p.Replicas {
+		res <- p.Replicas[i]
+	}
+	return nil
+}
+func resolveDynamodbTableReplicaGlobalSecondaryIndexes(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.ReplicaDescription)
+	if len(r.GlobalSecondaryIndexes) == 0 {
+		return nil
+	}
+
+	val := make([]map[string]interface{}, len(r.GlobalSecondaryIndexes))
+	for i := range r.GlobalSecondaryIndexes {
+		val[i] = map[string]interface{}{
```

Comment:
Question: Keep this or make it a map of Name vs. Object(name,prov_throughput_override)? Order shouldn't matter anyway.

Reply:
map could be useful, but for me it's fine either way